### PR TITLE
Re-introduce exiting the runloop exiting when events end

### LINKED ISSUE #2225: Mobile simulator does not close
### Describe the bug:

On develop branch (for Aberlour), pressing the close button on the mobile simulator windows does not close it. This seems to be a regression as it works as expected in the 2.0.x point releases.

### To Reproduce:
Steps to reproduce the behaviour:
1. Open the mobile simulator (in fyne_demo or any other code of your choice).
2. Try to close the windows using "x" in the corner.

### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** 5.11.12
 - **Go version:** 1.16.4
 - **Fyne version:** develop @ https://github.com/fyne-io/fyne/commit/9f38927599d0339ab5d42e6a422e962618d5f5d9

### PR DESCRIPTION
**Pull request by andydotxyz**
Fixes #2225

### Checklist:

- [ ] Tests included. <- mobile runloop still one of our not-unit-tested, just run and close on x11...
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ More detailed release notes can be found on the [releases page](https://github.c
 * Sequential writes to preferences does not save to file (#2449)
 * Correct Android keyboard handling (#2447)
 * MIUI-Android: The widget’s Hyperlink cannot open the URL (#1514)
+* Mobile simulator does not close (#2225)
```
Comment (Jacalz) [inline CHANGELOG.md:68]: This was never an issue in earlier versions before 2.1. I’m not sure it should be in the release notes.

**Review by Jacalz — CHANGES_REQUESTED** [review]
Just one thought about the changelog. Otherwise, it looks good :)

File: CHANGELOG.md (hunk 1 above)
Comment (andydotxyz) [reply CHANGELOG.md:68]: Good point

**Review by Jacalz — APPROVED** [review]
Works like a charm. Nice work :)

**Pull request merged**